src/main/kotlin/com/exactpro/th2/check1/rule/sequence/NoMessageCheckTask.kt

```diff
@@ -94,13 +95,14 @@ class NoMessageCheckTask(
             matchNames = false,
             significant = false
         )
-        val comparisonContainer = ComparisonContainer(messageContainer, protoPreMessageFilter, result)
-        if (comparisonContainer.fullyMatches) {
-            preFilterMessagesCounter++
-            preFilterEvent.messageID(comparisonContainer.protoActual.metadata.id)
-        } else {
-            extraMessagesCounter++
-            resultEvent.messageID(comparisonContainer.protoActual.metadata.id)
+        messageContainer.protoMessage.metadata.run {
```

Comment:
When you don't want to return anything it is better to use `apply` (or `also`) instead
```suggestion
        messageContainer.protoMessage.metadata.apply {
```